shpc/main/registry/provider.py

```diff
@@ -32,24 +34,6 @@ class Provider:
     A general provider should retrieve and provide registry files.
     """
 
-    def __init__(self, source, *args, **kwargs):
-        if not (source.startswith("https://") or os.path.exists(source)):
-            raise ValueError(
-                "Registry source must exist on the filesystem or be given as https://."
-            )
-        self.source = source
-
-    def exists(self, name):
```

Comment:
What do you see as being the intended future use for this class? It seems so abstract and slimmed down I'm not sure it provides anything useful.

Reply:
I couldn't find anything in common between the implementations of the two sub-classes. There could still be a base class, but I think it should list all the methods that are expected from the sub-classes, and throw a "Not implemented" error, rather than doing `pass`

Reply:
Agree - that's what I typically do.